Separate core_ext

ColinDKelley opened the pull request.

ColinDKelley commented on lib/humanize/module.rb:
```diff
@@ -117,15 +118,3 @@ def initialize
     end
   end
 end
-
```
The big change is moving this out to `core_ext.rb`.

ColinDKelley commented on lib/humanize/core_ext.rb:
```diff
@@ -0,0 +1,20 @@
+module Humanize
+  def humanize(locale: Humanize.config.default_locale,
+               decimals_as: Humanize.config.decimals_as)
```
This is the glue code that converts from implicit `self` arg to explicit `self` passed as the `number` arg.

radar commented on lib/humanize/module.rb:
```diff
@@ -0,0 +1,120 @@
+      when :tr
+        [Humanize::Tr, ' ']
+      when :jp
+        [Humanize::Jp, '']
```
This is missing the recent zh-TW addition.

ColinDKelley replied:
@radar Oops, I didn't mean to make any changes to this locale list. I just looked over in `master` and didn't find any reference to `zh`. Can you point me to it?

And I'm looking at the build failures.

ColinDKelley replied:
The build failures were rubocop suggestions. All addressed here: https://github.com/radar/humanize/pull/88/commits/0abcfc97e926fbc5c9d0a6a93f573ad3d362ea82

radar replied:
> @radar Oops, I didn't mean to make any changes to this locale list. I just looked over in `master` and didn't find any reference to `zh`. Can you point me to it?
> 
> And I'm looking at the build failures.

Getting ahead of myself. It's in https://github.com/radar/humanize/pull/87.

radar replied:
Ok, merged #87. So can we now get that in here? I think it's just the two lines for the when + the config.

ColinDKelley replied:
Ok, I've merged latest `master` with that PR here.

radar reviewed (changes requested):
Please restore zh-TW support in `for_locale`.

ColinDKelley commented:
@radar BTW I think the `# frozen_string_literal: true` magic comment can help across the board here. I won't clutter this PR with that, but once this is merged, I'll submit another PR that includes that magic comment and then drops `.freeze` after all the string constants.

radar commented:
Fine by me regarding FSL.

Rubocop is being a grump. Would you mind fixing that up? I can't push to your branch due to permission restrictions.

ColinDKelley commented:
> Rubocop is being a grump. Would you mind fixing that up? I can't push to your branch due to permission restrictions.

Ah, I'd made this change suppress that rubucop failure but lost it in the merge-up. Merged back here: https://github.com/radar/humanize/pull/88/commits/a347bbb140a6a2d1f1af643c13e6a771e1b57b60

Also, I found I'd missed this recursive call that was depending on the core `Integer` class being extended. I've switched it to functional (which will always work): https://github.com/radar/humanize/pull/88/commits/9339246a259e90b046248080726e43e89e496137

And finally, I updated the README to document the new functional usage: https://github.com/radar/humanize/pull/88/commits/32bd066d754d4e04b4ad431ff9268ee4d49d7411

radar commented:
All checks out to me. Thanks very much! ❤️
